content/blog/2024/12/how-to-migrate-from-idf-bootloader-to-mcuboot-using-esp-self-reflasher/index.md

```diff
@@ -0,0 +1,271 @@
+
+```
+http://<HOST_IP>:<HOST_PORT>/mcuboot-esp32.bin
+```
+
+Also configure the `Bootloader region address` (the Bootloader offset mentioned above) and the `Bootloader region size`.
+
+Then configure the similar parameters on the `Example Configuration -> Application reflash configuration` menu, set the `Application upgrade url endpoint` (e.g. `http://192.168.0.100:8070/app_upd.bin`):
+
+```
+http://<HOST_IP>:<HOST_PORT>/app_upd.bin
+```
+
+Set the `Application region address` and `Application region size`. Note that if the `boot_swap_download_example` is flashed standalone directly as is, these must match the image Secondary Slot from the target MCUboot image organization as the example expects MCUboot to do the swapping to the Primary Slot later, otherwise, if the example is OTA upgraded which this tutorial walk through, these configs must match Primary Slot. It will depend on where the reflashing application is running from (Factory partition or OTA partition), as it should not overwrite itself on the process.
+
+Configure the preferred network connection settings on the menu `Example Connection Configuration`, such as Wi-Fi SSID and Password.
+
+Save and exit the menuconfig.
+
+Build the example:
+
+```bash
+idf.py build
+```
+
+#### Example 2: Embedded the Target Reflashing Image
+
+Alternatively, ESP Self-Reflasher component can be used without network connection and download step, however as using this way the "bootloader + target reflashing image" will be embedded on the Reflashing Application, it must not exceed the total size that the first OTA supports.
+
+If the constraint is not a problem, `boot_swap_embedded_example` can be used as start point for the Reflashing application. Note that for this guide, the size of the OTA partitions on the partition table may be changed (see next section).
+
+```bash
+cd <IDF_DIR>/components/esp-self-reflasher/examples/boot_swap_embedded_example/
+```
+
+In this example, the MCUboot bootloader and target reflashing image needs to be merged in one binary as it goes embedded to the application.
+
+```bash
+esptool.py -c esp32 merge_bin --output <IDF_DIR>/components/esp-self-reflasher/examples/boot_swap_embedded_example/example_bin_dir/app_upd.merged.bin 0x0000 <NUTTX_DIR>/mcuboot-esp32.bin 0xF000 <NUTTX_DIR>/nuttx.bin
+```
+
+**Note** that in ESP32 case, the offsets on the merge step are shifted minus 0x1000 because of its bootloader offset 0x1000.
+
+Configure the example:
+
+```bash
+idf.py menuconfig
+```
+
+Enter menu `Example Configuration` and set the directory where the merged binary is placed and its name. Set the `Destination address`, that for this example will be the Bootloader offset, and the `Destination region size`, that must be the image slot size from the target flash organization plus the bootloader region size.
+
+Save and exit the menuconfig.
+
+Build the example:
+
+```bash
+idf.py build
+```
+
+### Building and flashing the IDF application
```

Comment:
The same here, looks like this section has a procedure or a how-to, please format it as a procedure with distinct numbered steps and indented code blocks to show that they belong to this particular step.